unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -5124,106 +5092,99 @@ private IEnumerator checkDropHandObjectActionFast(SimObjPhysics currentHandSimOb
+                return null;
             }
 
-            //an objectId was given, so find that target in the scene if it exists
-            else
-            {
-                if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(action.objectId)) {
-                    errorMessage = "Object ID appears to be invalid.";
-                    actionFinished(false);
-                    return;
-                }
-                
-                //if object is in the scene and visible, assign it to 'target'
-                foreach (SimObjPhysics sop in VisibleSimObjs(action)) 
-                {
-                    target = sop;
-                }
+            // if object is in the scene and visible, assign it to 'target'
+            SimObjPhysics target = null;
+            foreach (SimObjPhysics sop in VisibleSimObjs(objectId: objectId, forceVisible: forceAction)) {
+                target = sop;
             }
+            return target;
+        }
 
-            if (target) {
-
-                if (!action.forceAction && target.isInteractable == false) {
-                    errorMessage = "object is visible but occluded by something: " + action.objectId;
-                    actionFinished(false);
-                }
-
-                if (target.GetComponent<CanOpen_Object>()) {
-                    CanOpen_Object codd = target.GetComponent<CanOpen_Object>();
-
-                    //if object is open, close it
-                    if (codd.isOpen) {
-                        // codd.Interact();
-                        // actionFinished(true);
-                        StartCoroutine(InteractAndWait(codd, action.simplifyPhysics));
-                    } else {
-                        errorMessage = "object already closed: " + action.objectId;
-                        actionFinished(false);
-                    }
-                }
+        // Helper method that parses (x and y) parameters to return the
+        // sim object that they target.
+        private SimObjPhysics getTargetObject(
+            float x,
+            float y,
+            bool forceAction
+        ) {
+            // no target object specified, so instead try and use x/y screen coordinates
+            SimObjPhysics target = null;
+            if(!ScreenToWorldTarget((float) x, (float) y, ref target, !forceAction)) {
+                // error message is set insice ScreenToWorldTarget
+                return null;
+            }
+            return target;
+        }
 
-                else
-                {
-                    errorMessage = "target must be Openable to close";
-                    actionFinished(false);
-                    return;
-                }
+        // syntactic sugar for open object with openness = 0.
+        public void CloseObject(
+            string objectId,
+            bool forceAction = false
+        ) {
+            OpenObject(objectId: objectId, forceAction: forceAction, openness: 0);
+        }
```

Comment:
`CloseObject` is now treated as syntactic sugar for `OpenObject`, with `openness` being 0. Previously, it ran its own checks that were identical to the checks ran within `OpenObject`.

Reply:
Looks good. This also helps with backwards compatibility right? I think there is a discussion to be had to completely remove CloseObject and simply have people use OpenObject with `openness` at 0, but I'm not sure which is more intuitive.

Reply:
Yep: backward compatible. Agree that since `CloseObject` is so cheap now, it would add nothing to remove it.